.claude/hooks/detect_speculation_and_fake_code.sh

```
@@ -22,26 +22,27 @@ if PROJECT_ROOT_FROM_GIT=$(git rev-parse --show-toplevel 2>/dev/null); then
     else
         echo "Warning: Cannot determine project root and fallback path invalid" >&2
         exit 0
     fi
 fi
 
 # SECURITY: Early path validation to prevent attacks
-# Enhanced path validation - reject dangerous patterns (only actual traversal sequences)
+# Enhanced path validation - reject dangerous patterns (including trailing traversal)
 case "$PROJECT_ROOT" in
-    *'/../'*|*'/..')
+    *'/../'*|*'/..'|*'/../')
```

Shellcheck (0.11.0)

[warning] This pattern always overrides a later one on line 41.

(SC2221)